cmd/krew/cmd/list.go

```diff
@@ -56,17 +136,44 @@ Remarks:
 				return nil
 			}
 
-			// print table
-			var rows [][]string
-			for p, version := range plugins {
-				rows = append(rows, []string{p, version})
+			if *output.OutputFormat == "wide" || (*output.OutputFormat == "" && isTerminal(os.Stdout)) {
+				// print table
+				var rows [][]string
+				for p, version := range plugins {
+					rows = append(rows, []string{p, version})
+				}
+				rows = sortByFirstColumn(rows)
+				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
-			rows = sortByFirstColumn(rows)
-			return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+
+			if slice.ContainsString(output.AllowedFormats(), *output.OutputFormat, nil) {
+				objs := make([]Entry, 0, len(plugins))
+				for plugin, version := range plugins {
+					obj := Entry{plugin, version}
+					objs = append(objs, obj)
+				}
+				objs = sortByName(objs)
+				for _, obj := range objs {
+					p, err := output.ToPrinter()
```

Comment:
I think the proper way is to instantiate the printer once outside this for loop and reuse it for every item.

Reply:
Definitely much better, committing the change now.